src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -166,9 +166,30 @@ private protected void SuppressIsEnabled(bool suppress)
 			}
 		}
 
-		private object CoerceIsEnabled(object baseValue)
+		private protected virtual object CoerceIsEnabled(object baseValue, DependencyPropertyValuePrecedences precedence)
 		{
-			return _suppressIsEnabled ? false : baseValue;
+			if (_suppressIsEnabled)
+			{
+				return false;
+			}
+
+			// The baseValue hasn't been set inside PropertyDetails yet, so we need to make sure we're not
+			// reading soon-to-be-outdated values
+
+			var parentValue = precedence == DependencyPropertyValuePrecedences.Inheritance ?
+				baseValue :
+				this.GetValue(IsEnabledProperty, DependencyPropertyValuePrecedences.Inheritance);
+
+			// If the parent is disabled, this control must be disabled as well
+			if (parentValue != DependencyProperty.UnsetValue && !(bool)parentValue!)
```

Comment:
Let's not assume that parentValue is a boolean, it may happen as a result of invalid binding by the user. Let's use `parentValue as bool parentValueAsBool`.